Formula/a/azqr.rb

```diff
@@ -0,0 +1,20 @@
+class Azqr < Formula
+  desc "Azure Quick Review"
+  homepage "https://azure.github.io/azqr"
+  url "https://github.com/Azure/azqr.git",
+    tag:      "v.2.0.3",
+    revision: "655239455eec8ac434b9ebc7a68af9c2b117499b"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w -X 'github.com/Azure/azqr/cmd/azqr.version=#{version}'"), "./cmd/main.go"
+
+    generate_completions_from_executable(bin/"azqr", "completion")
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/azqr -v")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I've added an additional test that will start with a file that won't exist. 

Question to you @SMillerDev, are tests run on every machine or just while CI/CD? I was afraid to add an actual test that uses or tries using authentication because it could be that people have az cli logged in. 
If it's only on an agent I can add that too.

Thanks in advance!

Reply:
I added the auth failure test already while waiting for your response!